src/tty.c

```diff
@@ -152,83 +169,23 @@ void TTY_cmd() {
 
     DLTTYPortCmd->command &= ~PUT_CHAR;
   }
-} /* TTY_cmd end */
-
-void tty_setparam() {
-#ifdef TRACE
-  printf("TRACE: tty_setpram()\n");
-#endif
-
-  if (DLTTYPortCmd->outdata & SET_BAUD_RATE) tty_setbaudrate();
-
-} /* tty_setpram end */
-
-void tty_setbaudrate() {
-  char baudrate;
-  int stat;
-
-#ifdef TRACE
-  printf("TRACE: tty_setbaudrete()\n");
-#endif
-
-  if (TTY_Fd >= 0) {
-    if ((baudrate = tty_baudtosymbol(DLTTYOut->line_speed)) != -1) {
-      TTY_Mode.sg_ispeed = baudrate;
-      TTY_Mode.sg_ospeed = baudrate;
-      stat = ioctl(TTY_Fd, TIOCSETP, &TTY_Mode);
-    } else {
-      error("TTY: tty_setbaudrate");
-    }
-  }
-} /* tty_setbaudrate end */
-
-int tty_baudtosymbol(short aBaud)
-{
-#ifdef TRACE
-  printf("TRACE: tty_baudtosymbol(%x)\n", aBaud);
-#endif
-
-  if (aBaud == 0) return (B50);
-  if (aBaud == 1) return (B75);
-  if (aBaud == 2) return (B110);
-  if (aBaud == 3) return (B134);
-  if (aBaud == 4) return (B150);
-  if (aBaud == 5) return (B300);
-  if (aBaud == 6) return (B600);
-  if (aBaud == 7) return (B1200);
-  if (aBaud == 10) return (B2400);
-  if (aBaud == 12) return (B4800);
-  if (aBaud == 14) return (B9600);
-  if (aBaud == 15) return (EXTA);
-  return (-1);
+}
 
-} /* tty_baudtosymbol */
-
-void tty_debug(char *name)
+void tty_debug(const char *name)
 {
-  int stat;
-  struct sgttyb mode;
-
   printf("DEBUG: %s\n", name);
   printf("DEBUG: \t\tTTY_Dev            = \"%s\"\n", TTY_Dev);
   printf("DEBUG: \t\tTTY_Fd             = %d\n", TTY_Fd);
 
   if (TTY_Fd >= 0) {
-    stat = ioctl(TTY_Fd, TIOCGETP, &mode);
-    printf("DEBUG: \t\tTTY_Mode.sg_ispeed = %#x\n", mode.sg_ispeed);
-    printf("DEBUG: \t\tTTY_Mode.sg_ospeed = %#x\n", mode.sg_ospeed);
-    printf("DEBUG: \t\tTTY_Mode.sg_erase  = %#x\n", mode.sg_erase);
-    printf("DEBUG: \t\tTTY_Mode.sg_kill   = %#x\n", mode.sg_kill);
-    printf("DEBUG: \t\tTTY_Mode.sg_flags  = %#x\n", mode.sg_flags);
+    struct termios attr;
+    tcgetattr(TTY_Fd, &attr);
+    printf("DEBUG: \t\tTTY_Mode.sg_ispeed = %#x\n", cfgetispeed(&attr));
```

Comment:
Couple of printf format warnings here because `speed_t` is `unsigned int` on FreeBSD (Linux?) but `unsigned long` on macOS.  Probably reasonable to cast to `unsigned int` since we're talking 0..38400.

Reply:
Fixed and force pushed.